components/registry-facade/pkg/registry/blob.go

```diff
@@ -334,3 +348,55 @@ func (pbs *configBlobSource) getConfig(ctx context.Context) (rawCfg []byte, err
 	rawCfg, err = json.Marshal(cfg)
 	return
 }
+
+type ipfsBlobSource struct {
+	source *IPFSBlobCache
+}
+
+func (sbs ipfsBlobSource) HasBlob(ctx context.Context, spec *api.ImageSpec, dgst digest.Digest) bool {
+	_, err := sbs.source.Redis.Get(ctx, dgst.String()).Result()
+	return err == nil
+}
+
+func (sbs ipfsBlobSource) GetBlob(ctx context.Context, spec *api.ImageSpec, dgst digest.Digest) (mediaType string, url string, data io.ReadCloser, err error) {
+	log := log.WithField("digest", dgst)
+
+	var ipfsCID string
+	ipfsCID, err = sbs.source.Redis.Get(ctx, dgst.String()).Result()
+	if err != nil {
+		log.WithError(err).Error("unable to get blob details from Redis")
+		err = distv2.ErrorCodeBlobUnknown
+		return
+	}
+
+	ipfsFile, err := sbs.source.IPFS.Unixfs().Get(ctx, icorepath.New(ipfsCID))
+	if err != nil {
+		log.WithError(err).Error("unable to get blob from IPFS")
+		err = distv2.ErrorCodeBlobUnknown
+		return
+	}
+
+	f, ok := ipfsFile.(interface {
+		files.File
+		io.ReaderAt
+	})
+	if !ok {
+		log.WithError(err).Error("IPFS file does not support io.ReaderAt")
+		err = distv2.ErrorCodeBlobUnknown
+		return
+	}
+
+	mediaType, err = sbs.source.Redis.Get(ctx, mediaTypeKeyFromDigest(dgst)).Result()
+	if err != nil {
+		log.WithError(err).Error("cannot get media type from Redis")
+		err = distv2.ErrorCodeBlobUnknown
+		return
+	}
+
+	log.Debug("returning blob from IPFS")
+	return mediaType, "", f, nil
```

Comment:
This would do away the need for `nerdctl-registry`, right?
If so, we should either remove the `manifestModifier` based IPFS path, or make it configurable, so that IPFS in registry-facade can be used without the `registry` API in containerd. I'd vote for removing it.

Reply:
> This would do away the need for nerdctl-registry, right?

Yes.

Reply:
Removed